design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+		- [IncludeClusterResources change](#includeclusterresources-change)
+	- [Implementation](#implementation)
+	- [Open Issues](#open-issues)
+
+## Abstract
+The current filter (IncludedResources/ExcludedResources + IncludeClusterResources flag) is not enough for some special cases, e.g. all namespaced resources + some kind of cluster resource and all namespaced resources + cluster resource excludes.
+Propose to separate `IncludedResources` into cluster scope and namespace scope parameters, and same change for `ExcludedResources`.
+
+## Background
+There are two sets of resource filters for Velero: `IncludedNamespaces/ExcludedNamespaces` and `IncludedResources/ExcludedResources`. 
+`IncludedResources` means only including the resource types specified in the parameter.
+The k8s resources are separated into cluster scope and namespaced scope.
+As a result, it's hard to include all resources in one group and only including specified resource in the other group.
+
+## Goals
+- Make the existing resource filters supporting more scenarios without conflict.
+
+## Non Goals
+- Enrich the resource filtering rules.
+
+
+## High-Level Design
+Four new parameters are added into command `velero backup create`: `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources`. 
+`--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` are used to filter cluster scope resources included or excluded in backup per resource type.
+`--include-namespaced-resources` and `--exclude-namespaced-resources` are used to filter namespace scope resources included or excluded in backup per resource type.
+
+### Parameters Rules
+
+* `--include-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources` valid value include `*` and comma separated string.
+
+* `--exclude-namespaces` also supports `*`.
+
+* `--exclude-cluster-scope-resources` valid value is comma separated string.
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` can only contain cluster scope resource type names. If namespace scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` specified resources type lists, validation failure should be returned.
+
+* `--include-namespaced-resources` and `--exclude-namespaced-resources` can only contain namespace scope resource type names. If cluster scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-namespaced-resources` and `--exclude-namespaced-resources` specified resources type lists, validation failure should be returned.
+
+* If  `--include-namespaced-resources` is not present, it means including all namespace scope resources per resource type.
```

Comment:
I think this one should probably function like include-cluster-scope-resources for consistency. If not present, none included. Users should be required to include it with "*" -- this is a change from behavior of the other include-resources field, but that way with the new way of specifying resources, cluster-scoped and namespaced resources are handled in the same way to avoid confusion.